Release/include/pplx/pplxtasks.h

```diff
@@ -1866,7 +1866,7 @@ class task_completion_event
         {
             ::pplx::extensibility::scoped_critical_section_t _LockHolder(_M_Impl->_M_taskListCritSec);
 
-            if (!_IsTriggered())
+            if (!_IsTriggered() && !_M_Impl->_HasUserException())

```

Comment:
is this something we should push upstream to pplx?

Reply:
Yeah, I was thinking about that... Seems like the right thing to do 😄 I'll put together a PR for it.

Reply:
Submitted: https://github.com/microsoft/cpprestsdk/pull/1146